test/functional/test_framework/test_node.py

```diff
@@ -185,10 +184,8 @@ def p2p(self):
     def disconnect_p2ps(self):
         """Close all p2p connections to the node."""
         for p in self.p2ps:
-            # Connection could have already been closed by other end.
-            if p.connection is not None:
-                p.connection.disconnect_node()
-        self.p2ps = []
+            p.peer_disconnect()
+        del self.p2ps[:]
```

Comment:
stylte-nit: Any reason for this change? I think the previous is easier to read python code.

Reply:
There's some discussion of the differences here: https://stackoverflow.com/questions/850795/different-ways-of-clearing-lists

If there was a different reference to this array, then `self.p2ps = []` wouldn't clear that array, it would just change `self.p2ps` to point to a new, empty array.

Practically, there's not much difference, but I think I prefer `del self.p2ps[:]`. I actually got this from @ryanofsky's commit - perhaps he had a specific reason to make this change?

Reply:
Ok, fine to keep, since it is ever-so-slightly safer and already used by this syntax in `stop_node`

Reply:
Main reason for suggesting this is that having multiple instances of the same list can make things confusing when you are debugging.